dali_tf_plugin/dali_dataset_op.cc

```diff
@@ -274,10 +252,32 @@ class DALIDatasetOp : public DatasetOpKernel {
             enable_memory_stats_(enable_memory_stats) {}
 
       Status Initialize(IteratorContext* context) override {
-        if (!dataset()->fail_on_device_mismatch_) {
-            LOG(WARNING) << "DALI LOG: Allocator Name in Iterator: " << context->allocator({})->Name();
+        return CheckOutputDevices(); 
+      }
+
+      Status CheckOutputDevices() {
+        auto num_outputs = daliGetNumOutput(&pipeline_handle_);
+        for (auto i  = 0; i < num_outputs; ++i) {
+          auto dali_device_type = daliGetOutputDevice(&pipeline_handle_, i);
+
+          if (dali_device_type != dataset()->device_type_) {
+            auto msg = dali::make_string(
+              "TF device and DALI device mismatch. TF device: ",
+              (dataset()->device_type_ == device_type_t::CPU ? "CPU" : "GPU"),
```

Comment:
nitpick: Add ```
std::string to_string(device_type_t type)```

Reply:
Can you explain what do you mean here? I compare two `device_type_t` things and return `string` from ternary operator.